### PR TITLE
dnf5: Run transaction test for offline transactions

### PR DESCRIPTION
**Pull request by m-blaha**
A serialized offline transaction consists only of local packages
downloaded to the `<system state dir>/offline` location. Unfortunately,
PGP checks for these packages are disabled by default due to the
`localpkg_gpgcheck` option default value.
Instead of testing the serialized transaction, this patch performs an
RPM transaction test on the originally resolved transaction, which still
retains information about the repositories from which the packages were
downloaded.
This approach also saves time required for deserialization and resolving
the testing transaction.

Resolves: https://github.com/rpm-software-management/dnf5/issues/1668
Resolves: https://github.com/rpm-software-management/dnf5/issues/1667

Tests: https://github.com/rpm-software-management/ci-dnf-stack/pull/1549

### CODE REVIEW

**Comment by m-blaha** [comment]
The original approach had one advantage that the exactly same transaction that would be performed after `dnf offline reboot` was tested. Do you think it makes sense to at least try to resolve the serialized transaction (although it takes a long time so we should properly inform the user on what's going on). @evan-goode @jan-kolarik

**Review by evan-goode — APPROVED** [review]
I'm fine with this, I tested upgrading F39->F40 and it works.

> Do you think it makes sense to at least try to resolve the serialized transaction (although it takes a long time so we should properly inform the user on what's going on).

I think we can skip it. The main reason I implemented it this way was that it seemed to not be possible to test the (unserialized) offline transaction because the RPMs are all stored in the same directory. But that turned out to be due to a different bug. 

Assuming the transaction serialization logic is reliable, testing the serialized transaction is unnecessary and costs time.

File: dnf5/context.cpp [hunk 1]
```diff
@@ -321,6 +321,27 @@ void Context::Impl::load_repos(bool load_system) {
 }
 
 void Context::Impl::store_offline(libdnf5::base::Transaction & transaction) {
+    // Test the transaction
```
Comment (jrohel) [inline dnf5/context.cpp:324]: Scenario:
1. a new transaction is tested
2. `get_status()` of the offline transaction status is tested - if there is already an offline transaction in the queue and an `AbortedbyUserError` exception is thrown, the tested transaction is discarded
3. serialize the transaction under test

Can we change the order? Test new transaction only if `get_status()` test passes?
1. `get_status()` of offline transaction status is tested
2. a new transaction is tested
3. serialize the transaction under test

Comment (m-blaha) [reply dnf5/context.cpp:324]: That's a good point. Maybe we should do the status test even before downloading transaction packages.

Comment (m-blaha) [reply dnf5/context.cpp:324]: I've added a commit with such optimization. Now the offline transaction status is checked before the download phase.

**Pull request merged**